src/config/job-config/actions/corejobactioncreators.module.ts

```diff
@@ -0,0 +1,93 @@
+import { Module } from "@nestjs/common";
+import { LogJobActionCreator } from "./logaction/logaction.service";
+import { LogJobActionModule } from "./logaction/logaction.module";
+import { EmailJobActionCreator } from "./emailaction/emailaction.service";
+import { EmailJobActionModule } from "./emailaction/emailaction.module";
+import { actionType as logActionType } from "./logaction/logaction.interface";
+import { actionType as emailActionType } from "./emailaction/emailaction.interface";
+import { ValidateJobActionModule } from "./validateaction/validateaction.module";
+import { actionType as validateActionType } from "./validateaction/validateaction.interface";
+import {
+  ValidateCreateJobActionCreator,
+  ValidateJobActionCreator,
+} from "./validateaction/validateaction.service";
+import { URLJobActionModule } from "./urlaction/urlaction.module";
+import { URLJobActionCreator } from "./urlaction/urlaction.service";
+import { actionType as urlActionType } from "./urlaction/urlaction.interface";
+import { RabbitMQJobActionModule } from "./rabbitmqaction/rabbitmqaction.module";
+import { actionType as rabbitmqActionType } from "./rabbitmqaction/rabbitmqaction.interface";
+import { RabbitMQJobActionCreator } from "./rabbitmqaction/rabbitmqaction.service";
+import {
+  CREATE_JOB_ACTION_CREATORS,
+  UPDATE_JOB_ACTION_CREATORS,
+} from "../jobconfig.interface";
+
+/**
+ * Provide a list of built-in job action creators.
+ *
+ * CREATE_JOB_ACTION_CREATORS and UPDATE_JOB_ACTION_CREATORS can be extended (eg by a
+ * plugin) with additional creators.
+ */
+@Module({
+  imports: [
+    EmailJobActionModule,
+    LogJobActionModule,
+    ValidateJobActionModule,
+    URLJobActionModule,
+    RabbitMQJobActionModule,
+  ],
+  providers: [
```

Comment:
I can see that each module in the array will be initialized two times. One for `CREAT_JOB_ACTION` and the other for `UPDATE_JOB_ACTION`.
What if I want to use only RabbitMQJobAction and do not need the others, how can I disable them from initializing?

Reply:
You could disable them by providing different arrays to nestjs for the `CREATE_JOB_ACTION_CREATORS` and `UPDATE_JOB_ACTION_CREATORS` tokens. Currently `app.module.ts` imports `CoreJobActionCreators` statically. However, if we implement a plugin mechanism in the future (#862) then I would expect RabbitMQJobAction to be moved from core to a plugin, and `AppModule` would use some kind of dynamic import mechanism. It was decided this was too complicated for now, so all actions are included in the core.

Reply:
If it takes much more time to implement then it's better to skip it for now and add it in the future.
NestJS recently introduced new concept of [Conditional module configuration](https://docs.nestjs.com/techniques/configuration#conditional-module-configuration) which might be useful for future reference